openstudiocore/src/utilities/filetypes/EpwFile.cpp

```diff
@@ -2645,6 +4477,48 @@ namespace openstudio{
     return true;
   }
 
+  bool EpwFile::parseDesignConditions(const std::string& line)
+  {
+    // DESIGN CONDITIONS,1,Climate Design Data 2009 ASHRAE Handbook,,Heating,12,-17.4,-14,-21.5,0.7,-11.7,-18.9,0.9,-6.9,14.1,1.8,12,2.4,3.3,160,Cooling,7,15.2,34.6,15.7,33.2,15.6,31.8,15.4,18.3,27.3,17.6,27,17,26.5,4.2,80,16,14,19.9,15.2,13.2,19.7,14.1,12.3,19.6,58.3,27,55.9,26.9,53.8,26.3,722,Extremes,11.9,10.4,8.8,20.7,-22.7,37.1,2.8,1.3,-24.7,38,-26.3,38.8,-27.9,39.5,-29.9,40.5
+    // DESIGN CONDITIONS,Number of Design Conditions,Title of Design Condition,Design Stat,HDB 99.6%,HDB 99%,X WS 1%,X WS 2.5%,X WS 5%,CM WS .4%,CM MDB .4%,CM WS 1%,CM MDB 1%,MWS 99.6%,PWD 99.6%,MWS .4%,PWD .4%,X MnDB Max,X MnDB Min,X StdDB Max,X StdDB Min,Design Stat,CDB .4%,C MWB .4%,CDB 1%,C MWB 1%,CDB 2%,C MWB 2%,E WB .4%,E MDB .4%,E WB 1%,E MDB 1%,E WB 2%,E MDB 2%,DP .4%,HR .4%,MDB .4%,DP 1%,HR 1%,MDB 1%,DP 2%,HR 2%,MDB 2%,DB Range
+    std::vector<std::string> split = splitString(line, ',');
+
+    if (split[0] != "DESIGN CONDITIONS") {
+      LOG(Error, "Missing DESIGN CONDITIONS specifier in EPW file '" << m_path << "'");
+      return false;
+    }
+    else if (split[1] == "0") {
+      LOG(Warn, "Appears there are no design condition fields in the EPW file '" << m_path << "'");
+      return true;
+    }
+
+    int nDesignConditions = std::stoi(split[1]);
+
+    if (split.size() < 70) {
+      LOG(Error, "Expected at least 70 design condition fields rather than the " << split.size() << " fields in the EPW file '" << m_path << "'");
+      return false;
+    }
+    else if (split.size() > 70) {
```

Comment:
Shouldn't you check that nDesignConditions  and split.size agree?

Reply:
Yes you should check them otherwise you'll get out of index errors below

Reply:
Will implement a check for making sure that nDesignConditions agrees with the expected length of the split array.